### ISSUE TITLE
Class-based precompilers are never cached when compression is disabled

### ISSUE THREAD

**Issue by Alex-Sichkar** [django-compressor/django-compressor#1269]
For example, the `django_libsass.SassCompiler` precompiler is never cached when `COMPRESS_ENABLED` is set to `False`, which is a common setting in development mode. This behavior results in significant delays in SCSS processing during development.

```python
COMPRESS_PRECOMPILERS = (
    ('text/x-scss', 'django_libsass.SassCompiler'),
)
COMPRESS_CACHEABLE_PRECOMPILERS = (
    'text/x-scss',
)
COMPRESS_ENABLED = False
```

Based on the source code, caching is only applied to precompilers defined as command-line commands. If a precompiler is a filter class (like SassCompiler), it is never cached:
https://github.com/django-compressor/django-compressor/blob/8795cf8549759e3ef78c24e8c0d8e60c57d5dee0/compressor/base.py#L323

I use `django_libsass.SassCompiler` to compile Bootstrap SCSS files, which takes around 500ms. Since the precompiler is not cached, SCSS is recompiled on every request during development, leading to unnecessary delays. The only workaround I found is to set COMPRESS_ENABLED=True, but this is inconvenient for development due to issues like the inability to use source maps effectively.

I understand that in cases where SCSS files have imports, changes to the imported files might not be detected automatically. However, this is acceptable in most scenarios, as developers can manually restart the development server if needed.

**Comment by carltongibson** [comment]
This is kind of the inverse of #274. @diox's comment there about not being able to please everyone comes mind. 

I wonder if wrapping the SASS filter in the cached filter would allow you to use that? (Haven't tried)

Not sure it's feasible (or worth the disruption) to change this. What do the filter authors say on this? It seems conventional to NOT cache SASS because of the import issue. (I grant you want to, but if it's a niche desire...)

**Comment by Alex-Sichkar** [comment]
@carltongibson

> I wonder if wrapping the SASS filter in the cached filter would allow you to use that? (Haven't tried)

That would definitely be sufficient for my case. I tried using [dart-sass](https://github.com/sass/dart-sass) this way, and it worked just fine.

```python
COMPRESS_PRECOMPILERS = (
    ('text/x-scss', './node_modules/.bin/sass {infile} {outfile}'),
)
COMPRESS_CACHEABLE_PRECOMPILERS = (
    'text/x-scss',
)
```

> It seems conventional to NOT cache SASS because of the import issue

In use cases where you're customizing a large CSS framework, the compilation process is always time-consuming. Recompiling on every request doesn't seem reasonable, in my opinion.

That said, I can manage by keeping compression enabled. To enable source maps in the development environment, I remove `compressor.filters.cssmin.rCSSMinFilter`.

**Comment by Alex-Sichkar** [comment]
@carltongibson 

> What do the filter authors say on this?

In my opinion, the issue is not specific to this particular precompiler but applies to class-based precompilers in general - they are not cached. When you add a class-based precompiler to COMPRESS_CACHEABLE_PRECOMPILERS, you expect it to be cached. However, it doesn't work as expected, and there are no warnings or errors to indicate this. Moreover, the documentation does not mention this behavior.

**Comment by carltongibson** [comment]
@Alex-Sichkar Yes, I get that. My query was more about whether caching of Sass output is a common desire?
